core/src/main/java/org/ehcache/Ehcache.java

```diff
@@ -1026,42 +1026,16 @@ public void close() {
     statusTransitioner.close().succeeded();
   }
 
-  @Override
-  public Maintainable toMaintenance() {
-    final StatusTransitioner.Transition st = statusTransitioner.maintenance();
-    try {
-      final Maintainable maintainable = new Maintainable() {
-        @Override
-        public void create() {
-          Ehcache.this.create();
-        }
-
-        @Override
-        public void destroy() {
-          Ehcache.this.destroy();
-        }
-
-        @Override
-        public void close() {
-          statusTransitioner.exitMaintenance().succeeded();
-        }
-      };
-      st.succeeded();
-      return maintainable;
-    } catch (RuntimeException e) {
-      st.failed(e); // this throws
-    }
-    throw new AssertionError("Should not reach this line... ever!");
-  }
-  
-  void create() {
+  void checkMaintenance() {
     statusTransitioner.checkMaintenance();
-    // TODO figure out persistence and user managed caches
   }
 
-  void destroy() {
-    statusTransitioner.checkMaintenance();
-    // TODO figure out persistence and user managed caches
+  StatusTransitioner.Transition internalToMaintenance() {
```

Comment:
I am trying out with having the `PersistentUserManagedEhcache` delegate to `Ehcache` instead of extending it.